docs/src/main/asciidoc/security-overview-concept.adoc

```diff
@@ -2,264 +2,33 @@
-
-* If you need to introspect the JWT tokens remotely, you can use either `quarkus-oidc` or `quarkus-elytron-security-oauth2` because they support the verification of the opaque or binary tokens by using remote introspection.
-`quarkus-smallrye-jwt` does not support the remote introspection of both opaque or JWT tokens but instead relies on the locally available keys that are usually retrieved from the OpenID Connect provider.
-
-* `quarkus-oidc` and `quarkus-smallrye-jwt` support the injecting of JWT and opaque tokens into the endpoint code.
-Injected JWT tokens provide more information about the user. All extensions can have the tokens injected as `Principal`.
-
-* `quarkus-smallrye-jwt` supports more key formats than `quarkus-oidc`. `quarkus-oidc` uses only the JWK-formatted keys that are part of a JWK set, whereas `quarkus-smallrye-jwt` supports PEM keys.
-
-* `quarkus-smallrye-jwt` handles locally signed, inner-signed-and-encrypted, and encrypted tokens.
-While `quarkus-oidc` and `quarkus-elytron-security-oauth2` can also verify such tokens but treats them as opaque tokens and verifies them through remote introspection.
-
-* If you need a lightweight library for the remote introspection of opaque or JWT tokens, use `quarkus-elytron-security-oauth2`.
-
-[NOTE]
-====
-Your decision to choose whether to use opaque or JWT token format will be driven by architectural considerations.
-Opaque tokens tend to be much shorter than JWT tokens but need most of the token-associated state to be maintained in the provider database.
-Opaque tokens are effectively database pointers.
-JWT tokens are significantly longer than the opaque tokens but the providers are effectively delegating storing most of the token-associated state to the client by storing it as the token claims and either signing or encrypting them.
-====
-
-The following table provides a summary of the options for each authentication mechanism:
-
-|===
-| | quarkus-oidc| quarkus-smallrye-jwt | quarkus-elytron-security-oauth2
-
-|Requires Bearer JWT verification
-|Local verification or introspection
-|Local verification
-|Introspection
-|Requires Bearer opaque token verification
-|Introspection
-|No
-|Introspection
-|Refreshing `JsonWebKey` set for verifying JWT tokens
-|Yes
-|Yes
-|No
-|Represent token as `Principal`
-|Yes
-|Yes
-|Yes
-|Inject JWT as MP JSON Web Token (JWT)
-|Yes
-|Yes
-|No
-|Authorization Code Flow
-|Yes
-|No
-|No
-|Multi-tenancy
-|Yes
-|No
-|No
-|User info support
-|Yes
-|No
-|No
-|PEM key format support
-|No
-|Yes
-|No
-|SecretKey support
-|No
-|In JSON Web Key (JWK) format
-|No
-|Inner-signed and encrypted or encrypted tokens
-|Introspection
-|Local verification
-|Introspection
-|Custom token verification
-|No
-|With injected JWT parser
-|No
-|Accept JWT as a cookie
-|No
-|Yes
-|No
-|===
-
-[[identity-providers]]
```

Comment:
Note: All of this H2 section of security-overview-concept.adoc is now moved as-is into a new concept topic in this PR named security-identity-providers-concept.adoc.